service/header/interface.go

```diff
@@ -0,0 +1,26 @@
+package header
+
+import "context"
+
+// Subscriber encompasses the behavior necessary for a node to
+// subscribe/unsubscribe from new ExtendedHeader events from the
+// network.
+type Subscriber interface {
+	Subscribe(ctx context.Context) (<-chan *ExtendedHeader, error)
+	Unsubscribe() error
+}
+
+// Exchange encompasses the behavior necessary for a node to
+// request ExtendedHeaders and respond to ExtendedHeader requests
+// from the network.
+type Exchange interface {
+	RequestHeaders(ctx context.Context, request *ExtendedHeaderRequest) ([]*ExtendedHeader, error)
+	RespondToHeadersRequest(ctx context.Context, request *ExtendedHeaderRequest) error
+}
+
+// Store encompasses the behavior necessary to store and retrieve ExtendedHeaders
+// from a node's local storage.
+type Store interface {
```

Comment:
Store should not rely on Request, as this is not the network. We also should have the ability to get Headers by their hashes. Instead, the interface should be like this:
```go
// TODO: This might change in the future.
type Hash []byte

type Store interface {
	Get(context.Context, Hash) (*ExtednedHeader, error)
	GetMany(context.Context, []Hash) ([]*ExtendedHeader, error)
	GetByHeight(context.Context, int64) ([]*ExtendedHeader, error)
	GetRangeByHeight(ctx context.Context, from, to int64) ([]*ExtendedHeader, error))
	Put/Add(context.Context, *ExtendedHeader) error
	Put/AddMany(context.Context, []*ExtendedHeader) error
}
```

Reply:
I agree with not having the req type in the store interface. Hlib's proposed interface makes sense.